src/cmd/services/m3coordinator/ingest/write.go

```diff
@@ -496,7 +499,7 @@ func (d *downsamplerAndWriter) writeAggregatedBatch(
 		}
 
 		for _, dp := range value.Datapoints {
-			if value.Attributes.PromType != ts.PromMetricTypeUnknown {
+			if d.storeMetricType && value.Attributes.PromType != ts.PromMetricTypeUnknown {
```

Comment:
Question: What happens if `d.storeMetricType` is false and there is no `value.Attributes.M3Type` - won't that end up in the `unknown m3type` code path?

Or do we rely on the default value of `M3Type` being gauge to select gauge? (from the proto `M3_GAUGE = 0;`)

If we're relying on that behavior perhaps add a comment as such?

Reply:
I will check the logic here. My intention here was just to disable the logic which I have added here: https://github.com/m3db/m3/pull/2941/files#diff-bd39ac14b90a7e77e48b860a55e1927d98fd3378a61676da3b11c8b4a7698958R499